nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/TestRepository.java

```diff
@@ -444,6 +449,68 @@ public void itCanHandleComplexPropertiesWithJson() throws IOException {
         assertEquals("file", document.getTitle());
     }
 
+    /**
+     * Dates can be handled properly if and only if the date is given as {@link java.lang.String}.
+     */
+    @Test
+    public void itCanHandleDateAsString() {
+        GregorianCalendar calendar = new GregorianCalendar(2017, Calendar.MAY, 4, 3, 2, 1);
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSX");
+        calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
+        formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
+        String calendarStr = formatter.format(calendar.getTime());
+        assertEquals("2017-05-04T03:02:01.00Z", calendarStr);
+
+        Document birthday = new Document("Tom's Birthday", "Birthday");
+        birthday.set("dc:title", "Tom's Birthday");
+        birthday.set("bd:name", "Tom");
+        birthday.set("bd:date", calendarStr);
+        birthday = nuxeoClient.repository().createDocumentByPath("/", birthday);
+        assertEquals("Birthday", birthday.getType());
+        assertEquals("Tom", birthday.getPropertyValue("bd:name"));
+        assertEquals("2017-05-04T03:02:01.00Z", birthday.getPropertyValue("bd:date"));
+
+        calendar.add(Calendar.MONTH, 1);
+        calendarStr = formatter.format(calendar.getTime());
+        assertEquals("2017-06-04T03:02:01.00Z", calendarStr);
+        birthday.set("bd:date", calendarStr);
+        birthday = nuxeoClient.repository().updateDocument(birthday);
+        assertEquals("2017-06-04T03:02:01.00Z", birthday.getPropertyValue("bd:date"));
+
+        formatter.setTimeZone(TimeZone.getTimeZone("GMT+1:00"));
+        calendarStr = formatter.format(calendar.getTime());
+        assertEquals("2017-06-04T04:02:01.00+01", calendarStr);
+        birthday.set("bd:date", calendarStr);
+        birthday = nuxeoClient.repository().updateDocument(birthday);
+        assertEquals("2017-06-04T03:02:01.00Z", birthday.getPropertyValue("bd:date"));
```

Comment:
Which timezones are non-UTC timezones, could you give me an example?

By the way, the returned value always ends with `Z`, which is weird. The timezone is supposed to be same with the server's. In my case, it should be `+01` at the end.